qiskit/transpiler/target.py

```diff
@@ -1311,151 +1314,218 @@ def from_configuration(
-                        try:
-                            duration = instruction_durations.get(gate, edge, unit="s")
-                        except TranspilerError:
-                            duration = None
+                                duration = instruction_durations.get(
+                                    inst=name, qubits=list(qubits), unit="s", parameters=None
+                                )
+                            except TranspilerError:
+                                warnings.warn(
+                                    f"Duration of instruction {name} on qubits {qubits} is "
+                                    "not found in `InstructionDurations` passed, "
+                                    "falling back to `BackendProperties` to fetch duration",
+                                    RuntimeWarning,
+                                )
+                                duration = (
+                                    params["gate_length"][0] if "gate_length" in params else None
+                                )
+                        else:
+                            duration = params["gate_length"][0] if "gate_length" in params else None
+
+                        if prop_name_map[name] is None:
+                            prop_name_map[name] = {}
+
+                        prop_name_map[name][qubits] = InstructionProperties(
+                            error=error, duration=duration
+                        )
 
-                    if error is None and duration is None and calibration is None:
-                        gate_properties[edge] = None
-                    else:
-                        gate_properties[edge] = InstructionProperties(
-                            duration=duration, error=error, calibration=calibration
+                        if isinstance(prop_name_map[name], dict) and any(
+                            v is None for v in prop_name_map[name].values()
+                        ):
+                            # Properties provides gate properties only for subset of qubits
+                            # Associated qubit set might be defined by the gate config here
+                            logger.info(
+                                "Gate properties of instruction %s are not provided for every qubits. "
+                                "This gate is ideal for some qubits and the rest is with finite error. "
+                                "Created backend target may confuse error-aware circuit optimization.",
+                                name,
+                            )
+                except BackendPropertyError:
+                    # This gate doesn't report any property
+                    continue
+
+            # Measure instruction property is stored in qubit property
+            prop_name_map["measure"] = {}
+            for qubit_idx in range(num_qubits):
+                qubit_prop = backend_properties.qubit_property(qubit_idx)
+                duration = (
```

Comment:
The same logic must be applied here because instruction duration and inst map can be also another source of duration (this design itself is not good though).

Reply:
I was not sure if the docstrings should be changed or the logic itself should be changed. That's why I request your guidance :)
Applied suggestion :)

I request you to tell a bit more about the flaw in the design :)